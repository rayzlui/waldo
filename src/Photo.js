import React from 'react';
import PropTypes from 'prop-types';

import './index.css';

class Photo extends React.Component {
  //photoDOM only deals with the DOM stuff, we'll have an actual photo
  //class that will break down the info from the database (namely the photourl and tags)
  selectPhoto() {
    this.props.selectPhoto(this.props.id);
  }

  render() {
    let click = null;
    if (this.props.height > 1) {
      click = this.selectPhoto.bind(this);
    }

    return (
      <img
        src={this.props.photo}
        alt={'ohnoesy i broked'}
        onClick={click}
        style={{
          height: 500 / this.props.height,
          width: 800 / this.props.width,
          display: 'inline-block',
        }}
      />
    );
  }
}

export default Photo;

Photo.propTypes = {
  selectPhoto: PropTypes.func,
  height: PropTypes.number,
  photo: PropTypes.string,
  width: PropTypes.number,
  id: PropTypes.number,
};
