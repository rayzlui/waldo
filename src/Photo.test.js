
import React from 'react'
import Photo from "./Photo.js";  
import {shallow} from 'enzyme'

describe("Photo", ()=>{
    it("should display 'ohnoesy i broked' when no prop.photo is entered", () => {
        const component = shallow(<Photo/>)

        expect(component).toMatchSnapshot();
    });

    it("should display 'imgurl' when prop.photo = 'imgurl'", () => {
        const component = shallow(<Photo photo = {"imgurl"}/>);
        var photo = component.find('img')
        
        //check that img.src = imgurl
        expect(photo.props().src).toEqual("imgurl")
    });

    //check img.onclick = click
    it("should have click action when props.height == 2", ()=>{
        var mockFunction = jest.fn()
        const wrapper = shallow(<Photo height = {2} selectPhoto = {mockFunction}/>)
        var photo = wrapper.find('img').simulate('click')
        expect(mockFunction).toBeCalled()
    })

    //check img.onclick = null
    it("should not have click action when props.height == 1", ()=>{
        var  mockFunction = jest.fn()
        const wrapper = shallow(<Photo height = {1} selectPhoto = {mockFunction}/>)
        wrapper.find('img').simulate('click')
        expect(mockFunction).not.toBeCalled()    
    })

    //check height = 500
    it("should have height 500px when props.height == 1", ()=>{
        
        const wrapper = shallow(<Photo height = {1}/>)
        var photo = wrapper.find('img')
        expect(photo.props().style.height).toEqual(500)
    })

    //check height = 500/x

    it("should have height 250px when props.height == 2", ()=>{
        const wrapper = shallow(<Photo height = {2}/>)
        var photo = wrapper.find('img')
        expect(photo.props().style.height).toEqual(250)
    })

    //check width = 800
    it("should have height 800px when props.width == 1", ()=>{
        const wrapper = shallow(<Photo width = {1}/>)
        var photo = wrapper.find('img')
        expect(photo.props().style.width).toEqual(800)
    })

    //check width = 800/x
    it("should have height 400px when props.width == 2", ()=>{
        const wrapper = shallow(<Photo width = {2}/>)
        var photo = wrapper.find('img')
        expect(photo.props().style.width).toEqual(400)
    })

    
})