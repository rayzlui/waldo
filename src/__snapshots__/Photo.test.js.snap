// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Photo should display 'ohnoesy i broked' when no prop.photo is entered 1`] = `
<img
  alt="ohnoesy i broked"
  onClick={null}
  style={
    Object {
      "display": "inline-block",
      "height": NaN,
      "width": NaN,
    }
  }
/>
`;
